text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,201 @@
+  one in `<buildpack>/dependency/metadata`.
+4. If the `URI` and `SHA256` are missing from the metadata, the code will be
+  compiled with the code from `<buildpack>/dependency/compilation`.
+  - Test the compiled dependency using the test from `<buildpack>/dependency/test`
+  - Upload the dependency to the dependency GCP bucket
+  - Add the GCP bucket access URI and the SHA256 of the dependency to the
+    metadta `URI` and `SHA256` fields
+5. Update the `buildpack.toml` with the new versions and metadata
+
+This new plan eliminates extra steps of storing metadata in a file for later
+use, and then separately updating the dependencies on a timer.
+
+### Workflows and Actions
+Pending the Phase 1 RFC is accepted, and it's decided dependency logic will
+live alongside the buildpack, workflows will be rolled out to all of the
+buildpacks from
+[paketo-buildpacks/github-config](https://github.com/paketo-buildpacks/github-config)
+inside the `implementation/.github/workflows` location.
+
+Actions will live in the github-config repository as well, under a
+directory named `dependency`, inside of the `actions` directory.
+
+The github-config repository `CODEOWNERS` file will be updated to set the
+`@paketo-buildpacks/dependencies-maintainers` to the owners of the
+`actions/dependency` and `implementation/.github/workflows/<all
+dependency-related workflows>`.
+
+
+Each of the step s outlined in the overview will be converted into workflow
+steps.
+
+Workflow 1:
+- Picking up new versions with buildpack code will be a workflow step
+  It should run twice per day.
+- Figuring out which versions will be added to the `buildpack.toml` will be
+  rolled into an action that outputs versions.
+- A Github Actions [`matrix
+  strategy`](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy)
+  will be used to iterate over each version to be added, to generate metadata
+  using the code from within the buildpacks.
+- In the same matrix loop, dependency compilation will be kicked off with the
+  code from the buildpacks on the basis of whether the metadata contains the
+  `SHA256` and `URI` of the dependency.
+- The dependency will be uploaded via an action
+- The metadata is updated with the SHA256 and URI in a step
+- Updating the `buildpack.toml` with the new versions and metadata will also be
+  delegated to a new action
+- A pull request will be opened with the changes to the `buildpack.toml`
+
+All of these steps will run as a workflow, but can be run easily manually.
+Additionally, we could eventually create a script inside of the
+`<buildpack>/scripts` directory to perform the main steps of the
+`buildpack.toml` update process locally.
+
+- TODO: how to deal w multiple variants of the same dependency
+- TODO: how to isolate failures? Should we split it into jobs incase failures occur
+- TODO: manually running? can you do a manual dispatch
```

Comment:
Yes, this should be simple and we should document it. I think that this will need to be run manually if we're in a rush - e.g. trying to ship a high-priority fix.